### PR TITLE
Create Search Functions

### PR DESCRIPTION
**Pull request by IanDuncanT**
The following commands have been updated.

- python3 yapi.py config
Updated to add api url for packages. Delete config.ini and run the command to create new from default.ini

- python3 yapi.py search [regex]
Search repository and locally for files matching regex. *You must escape bash.* This means that all \ become \\ for use in bash.

### CODE REVIEW

**Comment by IanDuncanT** [comment]
This closes #11

File: config.ini [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 
-[CACHE]
+[Cache]
 keep_cache = True
 cache_location = scripts/
 
-[REMOTE]
+[Search]
+search_local = True
+packages_search_remote = https://api.github.com/repos/YetAnotherPackageInstaller/packages-darwin/contents/scripts?ref=master
```
Comment (Wabri) [inline config.ini:10]: you can set this with a null value and set up by configuration.py

Comment (Wabri) [reply config.ini:10]: Ok you use the default.ini to do this

File: search.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+def search(pattern, url, extension, local_boolean, local_dir):
```
Comment (Wabri) [inline search.py:1]: I'm honest, it took me a while to understand the last two arguments.

File: search.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    #Get local files
+    if local_boolean:
+        for file in os.listdir(local_dir):
+            if file.endswith(extension):
+                if file.replace('.sh', '') not in files:
+                    files.append(file.replace('.sh', ''))
+    #Search
+    r = re.compile(pattern)
+    result = list(filter(r.match, files))
```
Comment (Wabri) [inline search.py:35]: I dont understand the match method usage

File: search.py (hunk 2 above)
Comment (IanDuncanT) [reply search.py:1]: local_boolean is a boolean to check for local files. This is useful for installing shell files that were downloaded or placed in there from other sources. local_dir is the scripts/ which is changed from the config.ini file and is where local scripts are stored.

File: search.py (hunk 3 above)
Comment (IanDuncanT) [reply search.py:35]: re.compile makes the regex pattern from the input. The r.match takes that specific pattern and goes through the files list with filter and makes a new list that is assigned to result. This list is the result of the regex search.

File: config.ini (hunk 1 above)
Comment (IanDuncanT) [reply config.ini:10]: I will change default.ini and require python3 yapi.py config in the readme.

Comment (IanDuncanT) [reply config.ini:10]: Put python3 yapi.py config in install.sh

**Review by Wabri — APPROVED** [review]

**Pull request merged**